features/device_key/source/DeviceKey.cpp

```diff
@@ -249,14 +249,13 @@ int DeviceKey::generate_key_by_trng(uint32_t *output, size_t& size)
 
     trng_init(&trng_obj);
 
-    final_size = 0;
+    //final_size = 0;
```

Comment:
Please remove `final_size` entirely -- at the moment, it is uninitialized but assigned to `size` in the end, which doesn't have an effect to the caller because `size` is no longer a reference, but nonetheless undefined behaviour.